cheatsheets_excluded/Content_Security_Policy_Cheat_Sheet.md

```diff
@@ -1,203 +1,199 @@
--   **font-src** (d) - restricts from where the protected resource can load fonts
--   **img-src** (d) - restricts from where the protected resource can load images
--   **media-src** (d) - restricts from where the protected resource can load video, audio, and associated text tracks
--   **object-src** (d) - restricts from where the protected resource can load plugins
--   **script-src** (d) - restricts which scripts the protected resource can execute. Additional restrictions against, inline scripts, and eval. Additional directives in CSP2 for hash and nonce support
--   **style-src** (d) - restricts which styles the user may applies to the protected resource. Additional restrictions against inline and eval.
--   **default-src** - Covers any directive with *(d)*
--   **frame-src** - restricts from where the protected resource can embed frames. Note, deprecated in CSP2
--   **report-uri** - specifies a URL to which the user agent sends reports about policy violation
--   **sandbox** - specifies an HTML sandbox policy that the user agent applies to the protected resource. Optional in 1.0
+- `img-src` specifies the URLs that images can be loaded from.
+- `manifest-src` specifies the URLs that application manifests may be loaded from.
+- `media-src` specifies the URLs from which video, audio and text track resources can be loaded from.
+- `prefetch-src` specifies the URLs from which resources can be prefetched from.
+- `object-src` specifies the URLs from which plugins can be loaded from.
+- `script-src` specifies the locations from which a script can be executed from. It is a fallback directive for other script-like directives.
+  - `script-src-elem` controls the location from which execution of script requests and blocks can occur.
+  - `script-src-attr` controls the execution of event handlers.
+- `style-src` controls from where styles get applied to a document. This includes `<link>` elements, `@import` rules, and requests originating from a `Link` HTTP response header field.
+  - `style-src-elem` controls styles except for inline attributes.
+  - `style-src-attr` controls styles attributes.
 
-### New in CSP2
+## Document Directives
 
--   **form-action** - retricts which URLs can be used as the action of HTML form elements
--   **frame-ancestors** - indicates whether the user agent should allow embedding the resource using a frame, iframe, object, embed or applet element, or equivalent functionality in non-HTML resources
--   **plugin-types** - restricts the set of plugins that can be invoked by the protected resource by limiting the types of resources that can be embedded
--   **base-uri** - restricts the URLs that can be used to specify the document base URL
--   **child-src** (d) - governs the creation of nested browsing contexts as well as Worker execution contexts
+Document directives instruct the browser about the properties of the document to which the policies will apply to. 
```

Comment:
Miss the directive `disown-opener`: Ensures a resource will disown its opener when navigated to.

Reply:
Kindly link the mentioned directives as it does not exist. It is not even in the beta CSPv3.

Reply:
It's an experimental feature, based on [supports mattrix](https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP#Browser_compatibility) in fact we should not add it so very sorry, my mistake.